mayastor/src/bdev/nexus/nexus_bdev_rebuild.rs

```diff
@@ -236,13 +238,14 @@ impl Nexus {
 
         match job.state() {
             RebuildState::Completed => {
-                recovering_child.out_of_sync(false);
+                recovering_child.set_state(ChildState::Open);
+                NexusChild::save_state_change();
                 info!(
                     "Child {} has been rebuilt successfully",
                     recovering_child.name
                 );
 
-                assert_eq!(recovering_child.status(), ChildStatus::Online);
+                assert_eq!(recovering_child.state(), ChildState::Open);
```

Comment:
Looks like this assert isn't needed any more as the state has just been set on line 241.

Reply:
I've removed it